org-code-javabuilder/lib/src/main/java/org/code/javabuilder/ExceptionHandler.java

```diff
@@ -32,6 +32,18 @@ public void handle(Throwable e) {
       return;
     }
 
+    // We should also exit on an OutOfMemoryError.
+    // This means something is wrong with our memory usage
+    // and is not likely to be recoverable.
+    if (e instanceof OutOfMemoryError) {
+      final InternalServerException error =
+          new InternalServerException(InternalExceptionKey.UNKNOWN_ERROR, e);
```

Comment:
Should this be wrapped in a `FatalError` instead since we take similar steps to shut down the runtime? Don't think it makes any difference for logging purposes, but we could also maybe consolidate this block with the block above

Reply:
If we wrap in `FatalError` this error message is sent to users: `Exception message: java.lang.OutOfMemoryError: Metaspace`. If we keep it as-is we get the "we had an error on our side" message, which I think makes more sense for this scenario. So I'm incline to leave it as-is even though it's a bit repetitive.

Reply:
Cool gotcha, sounds good. That being said, I think it would probably make sense for FatalErrors to also display the "we had an error on our side" message, similar to InternalExceptions right? I think the error is being printed because we don't have handling for it client side. I can make a separate task to look into that.

Reply:
Yeah that would be good!